espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/models/ActiveInputSources.java

```diff
@@ -0,0 +1,33 @@
+package io.appium.espressoserver.lib.helpers.w3c.models;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException;
+
+/**
+ * Active Input Source defined in W3C spec
+ *
+ * (see https://www.w3.org/TR/webdriver/#terminology-0)
+ */
+public class ActiveInputSources {
+    private Map<String, InputSource> inputSources = new HashMap<>();
+
+    public void addInputSource(InputSource inputSource) throws InvalidArgumentException {
+        if (inputSource.getId() == null) {
+            throw new InvalidArgumentException("Input source is missing ID");
+        }
+        this.inputSources.put(inputSource.getId(), inputSource);
+    }
+
+    /**
+     * Remove an input source and also remove it from InputStateTable
+     * @param inputSource
+     */
+    public void removeInputSource(InputSource inputSource) {
+        if (this.inputSources.containsKey(inputSource.getId())) {
```

Comment:
`containsKey`  may be omitted

Reply:
It's checking if the map of input sources contains the input source with that key